tests/test_integration.py

```diff
@@ -169,3 +164,35 @@ def test_verify_requires_python() -> None:
         InvalidRequires, match="Expected requires_python to be a '>=' specifier"
     ):
         verify_requires_python("==3.10")
+
+
+@pytest.mark.parametrize(
+    "distribution,expected_packages",
+    [
+        ("pytz", ["pytz-stubs"]),
+        ("Pillow", ["PIL-stubs"]),
+        ("protobuf", ["google-stubs"]),
+        ("google-cloud-ndb", ["google-stubs"]),
+    ],
+)
+def test_pkg_data_top_level_packages(
+    distribution: str, expected_packages: list[str]
+) -> None:
+    pkg_data = build_wheel.collect_package_data(THIRD_PARTY_PATH / distribution)
+    assert pkg_data.top_level_packages == expected_packages
+
+
+@pytest.mark.parametrize(
+    "distribution,expected_packages",
+    [
+        ("pytz", ["pytz-stubs"]),
+        ("Pillow", ["PIL-stubs"]),
+        ("protobuf", ["google-stubs.protobuf"]),
+        ("google-cloud-ndb", ["google-stubs.cloud.ndb"]),
+    ],
+)
+def test_pkg_data_non_namespace_packages(
+    distribution: str, expected_packages: list[str]
+) -> None:
+    pkg_data = build_wheel.collect_package_data(THIRD_PARTY_PATH / distribution)
+    assert pkg_data.top_level_non_namespace_packages == expected_packages
```

Comment:
I just noticed, I think there's no test to validate that `py.typed` is added to the right location(s). Could we add unit tests for the `if metadata.partial` branch of `generate_setup_file` calling `add_partial_markers`?

Reply:
Good idea, but I think it makes more sense as part of #121, considering the current command/query mixture.